core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -1172,51 +1165,46 @@ object Pull extends PullLowPriority {
 
       viewL(stream) match {
         case tst: Translate[h, G, _] => // y = Unit
-          val view = contP.asInstanceOf[Cont[Unit, G, X]]
+          val translateRunner: Run[h, X, F[End]] = new TranslateRunner(tst.fk, getCont[Unit, G, X])
           val composed: h ~> F = translation.compose[h](tst.fk)
-          val translateRunner: Run[h, X, F[End]] = new TranslateRunner(tst.fk, view)
           go[h, X, End](scope, extendedTopLevelScope, composed, translateRunner, tst.stream)
 
-        case action: Action[G, X, y] =>
-          val view: Cont[y, G, X] = contP.asInstanceOf[Cont[y, G, X]]
-          action match {
-            case output: Output[_] =>
-              interruptGuard(scope, view)(
-                runner.out(output.values, scope, view(unit))
-              )
-
-            case fmout: FlatMapOutput[g, z, _] => // y = Unit
-              goFlatMapOut[z](fmout, view.asInstanceOf[Cont[Unit, g, X]])
-
-            case u0: Uncons[g, y] =>
```

Comment:
Well, we would need to add, again, the `g` extra variable and then the `asInstanceOf` calls, just to avoid a (rather unnecessary) compiler warning. Perhaps we can just suppress some compiler warnings?

Reply:
I'm fine with suppressed warnings if that's straightforward.